### PR TITLE
Changed the MLP template repo to become an actual template

### PR DESCRIPTION
**Pull request by jinzhang21**
## What changes are proposed in this pull request?
We have decided to provide both examples and templates for different purposes when working with MLP. Examples are used for beginners to learn about MLP concepts and basic usage. Templates are used for repeatedly create ML projects easily. This PR does the transformation. I'll wait until the example repo setup successfully before merging this change.

## How is this patch tested?
Manual.

We require adding a link to the [workflow](https://github.com/mlflow/mlflow/actions/workflows/pipeline-template.yml) for testing the [mlflow/mlp-regression-template](https://github.com/mlflow/mlp-regression-template) code path.

Run the workflow with repository name: `mlflow/mlp-regression-template` and the branch`(Example:<user-name>:feature)` that you are working with.
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/regression`: Sklearn regression pipeline example
- [ ] `area/build`: Build and test infrastructure for MLflow pipeline example

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — COMMENTED** [review]
@jinzhang21 We need to update the mlflow docs for MLP before merging this. For example, https://mlflow.org/docs/latest/pipelines.html#pipeline-templates links to this template and says that it's a "fully-functional pipeline". We'd want to change the phrasing and link to a separate example repo.

We'll also want to update the MLP submodule in MLflow to point to the example.

**Comment by jinzhang21** [comment]
> @jinzhang21 We need to update the mlflow docs for MLP before merging this. For example, https://mlflow.org/docs/latest/pipelines.html#pipeline-templates links to this template and says that it's a "fully-functional pipeline". We'd want to change the phrasing and link to a separate example repo.
> 
> We'll also want to update the MLP submodule in MLflow to point to the example.

Absolutely. I have created a [Jira](https://databricks.atlassian.net/browse/ML-25518) ticket to track the tasks you mentioned.

File: README.md
Comment (dbczumar) [inline README.md:46]: Reminder to rename to mlp:

```suggestion
please checkout [this](https://github.com/mlflow/mlp-regression-example) repository.
```

File: pipeline.yaml
Comment (dbczumar) [inline pipeline.yaml:16]: Nit: Is the indentation change intended?

File: pipeline.yaml
Comment (dbczumar) [inline pipeline.yaml:90]: To clarify, batch scoring is optional, so we're keeping this commented out?

File: pipeline.yaml
Comment (dbczumar) [inline pipeline.yaml:86]: To clarify, batch scoring is optional, so we're keeping this commented out?

File: profiles/databricks.yaml
Comment (dbczumar) [inline profiles/databricks.yaml:2]: Shall we include optional configs for tracking URI / artifact location?

File: requirements.txt
Comment (dbczumar) [inline requirements.txt]: These are MLP requirements included in `mlflow[pipelines]`, so I think we can remove them from the template

File: steps/train.py
Comment (dbczumar) [inline steps/train.py]:
```suggestion
    # FIXME::OPTIONAL: return a scikit-learn-compatible regression estimator with fine-tuned hyperparameters.
```

File: steps/transform.py
Comment (dbczumar) [inline steps/transform.py]: I think we should place this inside `transformer_fn()`.

```suggestion
# FIXME::OPTIONAL: return a scikit-learn-compatible transformer object.
```

File: pipeline.yaml
Comment (jinzhang21) [reply pipeline.yaml:16]: Yes. I want to make "NOTE" to stand out.

File: pipeline.yaml
Comment (jinzhang21) [reply pipeline.yaml:90]: If it's not commented out, users have to supply a value to `SCORED_OUTPUT_DATA_FORMAT` in profile, which essentially makes batch scoring required.

**Review by dbczumar — APPROVED** [review]
@jinzhang21 LGTM with a few small comments. Can we manually verify that, after setting all fixme:required fields, the pipeline works (I.e. we aren't missing anything)?

File: profiles/databricks.yaml
Comment (jinzhang21) [reply profiles/databricks.yaml:2]: On Databricks? Do you mean sending tracking info to a different shard? I'd say it's an advanced option. We can mention it in the user guide but I don't think it's necessary to list it here.

**Review by jinzhang21 — COMMENTED** [review]
Many thanks, @dbczumar !

File: notebooks/databricks.py
Comment (sunishsheth2009) [inline notebooks/databricks.py:37]: Nit: We should replicate this in the example as well. 
It is a much better experience to see this data before running all the other steps

File: pipeline.yaml
Comment (sunishsheth2009) [inline pipeline.yaml:78]: Is there a reason we need this if everything below is optional, can we comment this out? Do we need to fix mlflow code for this?

**Review by sunishsheth2009 — APPROVED** [review]

File: notebooks/databricks.py
Comment (jinzhang21) [reply notebooks/databricks.py:37]: SG. I'll do another update in the example repo.

File: pipeline.yaml
Comment (jinzhang21) [reply pipeline.yaml:78]: MLflow needs to be fixed with @prithvikannan 's PR. The template doesn't work as is. I'm assuming it'll be fixed soon.
We can comment `predict` step out but I want to leave it there to indicate such step actually exists.

**Comment by jinzhang21** [comment]
Thanks for the review, @sunishsheth2009 and @dbczumar . I have confirmed it works with knowing gaps that other PR will fix soon. I'll merge this as is and continue to improve it as necessary to unblock works after.

**Pull request merged**
